lib/protocol/sequences/ChangeUser.js

```diff
@@ -7,29 +7,28 @@ module.exports = ChangeUser;
 Util.inherits(ChangeUser, Sequence);
 function ChangeUser(options, callback) {
   Sequence.call(this, options, callback);
-
-  this._user          = options.user;
-  this._password      = options.password;
-  this._database      = options.database;
-  this._charsetNumber = options.charsetNumber;
-  this._currentConfig = options.currentConfig;
+  this._options = options;
 }
 
 ChangeUser.prototype.start = function(handshakeInitializationPacket) {
+  var options = this._options;
+
   var scrambleBuff = handshakeInitializationPacket.scrambleBuff();
-  scrambleBuff     = Auth.token(this._password, scrambleBuff);
+  scrambleBuff     = Auth.token(options.password, scrambleBuff);
 
   var packet = new Packets.ComChangeUserPacket({
-    user          : this._user,
+    user          : options.user,
     scrambleBuff  : scrambleBuff,
-    database      : this._database,
-    charsetNumber : this._charsetNumber
+    database      : options.database,
+    charsetNumber : options.charsetNumber
   });
 
-  this._currentConfig.user          = this._user;
-  this._currentConfig.password      = this._password;
-  this._currentConfig.database      = this._database;
-  this._currentConfig.charsetNumber = this._charsetNumber;
+  var currentConfig = options.currentConfig;
+  currentConfig.user          = options.user;
+  currentConfig.password      = options.password;
+  currentConfig.database      = options.database;
+  currentConfig.charsetNumber = options.charsetNumber;
+  currentConfig.modified      = true;
```

Comment:
I ran a v8 regression against this really quick and this is altering the hidden class during runtime. Please ensure that the objects build the correct hidden class from the get-go and don't alter it through runtime